src/Analyzer.Core.UnitTests/TemplateAnalyzerTests.cs

```diff
@@ -40,6 +41,51 @@ public void AnalyzeTemplate_ValidInputValues_ReturnCorrectEvaluations(string res
             Assert.AreEqual(expectedEvaluationPassCount, evaluationsWithResults.Count(e => e.Passed));
         }
 
+        [DataTestMethod]
```

Comment:
I don't see a testcase to validate that issue 39 is actually closed by this PR, for example with a template that has a parameter defined in the nested template that doesn't have the same name as a parameter in the main template

Reply:
added